Source/WebCore/rendering/svg/SVGTextLayoutEngine.cpp

```diff
@@ -266,17 +268,17 @@ void SVGTextLayoutEngine::finalizeTransformMatrices(Vector<InlineIterator::SVGTe
     if (textBoxes.isEmpty())
         return;
 
-    for (auto textBox : textBoxes) {
-        auto textBoxTransformation = m_chunkLayoutBuilder.transformationForTextBox(textBox);
-        if (textBoxTransformation.isIdentity())
-            continue;
+    auto boxesToTransform = textBoxes | std::views::filter([&](const auto& textBox) {
+        return !m_chunkLayoutBuilder.transformationForTextBox(textBox).isIdentity();
+    });
 
-        auto it = m_fragmentMap.find(makeKey(*textBox));
-        if (it != m_fragmentMap.end()) {
-            for (auto& fragment : it->value) {
+    for (const auto& textBox : boxesToTransform) {
```

Comment:
Although using `const auto&` is better than just using `auto`, I think the benefit here is not much. `TextBoxIterator` has just a raw pointer and an unsigned so it is cheap to make a copy.